storm-core/src/jvm/org/apache/storm/spout/ShellSpout.java

```diff
@@ -47,6 +50,8 @@
     private String[] _command;
     private Map<String, String> env = new HashMap<>();
     private ShellProcess _process;
+    private volatile boolean _running = true;
+    private volatile Throwable _exception;
```

Comment:
nit: this is only ever set to a RuntimeException already, and then it is wrapped in a RuntimeException.  It might be nice to just have it be a RuntimeException to begin with and then we can skip the wrapping.